src/test/java/com/squareup/kotlinpoet/LambdaTypeNameTest.kt

```diff
@@ -76,4 +76,26 @@ class LambdaTypeNameTest {
           returnType = Unit::class.asTypeName())
     }.hasMessageThat().isEqualTo("Parameters with default values are not allowed")
   }
+
+  @Test fun lambdaReturnType() {
+    val returnTypeName = LambdaTypeName.get(
+        parameters = *arrayOf(Int::class.asTypeName()),
+        returnType = Unit::class.asTypeName())
+    val typeName = LambdaTypeName.get(
+        parameters = *arrayOf(Int::class.asTypeName()),
+        returnType = returnTypeName)
+    assertThat(typeName.toString())
+        .isEqualTo("(kotlin.Int) -> (kotlin.Int) -> kotlin.Unit")
```

Comment:
Actually, since "the arrow notation is right-associative", `(Int) -> (Int) -> Unit` is equivalent to `(Int) -> ((Int) -> Unit)`

Reply:
I'd feel more comfortable if we emit parenthesis although I can't imagine this occurs anywhere in the wild...

Reply:
I'll add this